L1Trigger/TrackTrigger/python/ProducerSetup_cfi.py

```diff
@@ -96,14 +97,20 @@
     OuterRadius         = cms.double( 112.7              ), # outer radius of outer tracker in cm
     InnerRadius         = cms.double(  21.8              ), # inner radius of outer tracker in cm
     HalfLength          = cms.double( 270.               ), # half length of outer tracker in cm
-    MaxPitch            = cms.double(    .01             ), # max strip/pixel pitch of outer tracker sensors in cm
-    MaxLength           = cms.double(   2.5              ), # max strip/pixel length of outer tracker sensors in cm
-    TiltApproxSlope     = cms.double(   0.886            ), # 
-    TiltApproxIntercept = cms.double(   0.504            ), # 
-    MindPhi             = cms.double(  0.0001            ), # minimum representable stub phi uncertainty
-    MaxdPhi             = cms.double(  0.02              ), # maximum representable stub phi uncertainty
-    MindZ               = cms.double(  0.1               ), # minimum representable stub z uncertainty
-    MaxdZ               = cms.double( 30.                ), # maximum representable stub z uncertainty
+    #TiltApproxSlope     = cms.double(   0.886            ), #
+    #TiltApproxIntercept = cms.double(   0.504            ), #  
+    TiltApproxSlope     = cms.double(   1.02399          ), # 
+    TiltApproxIntercept = cms.double(  -0.279601         ), # 
+    MindPhi             = cms.double(   0.0001           ), # minimum representable stub phi uncertainty
+    MaxdPhi             = cms.double(   0.02             ), # maximum representable stub phi uncertainty
+    MindZ               = cms.double(   0.1              ), # minimum representable stub z uncertainty
+    MaxdZ               = cms.double(  30.               ), # maximum representable stub z uncertainty
+    Pitch2S             = cms.double(   0.009            ), # strip pitch of outer tracker sensors in cm
```

Comment:
these things should be read from the tracker geometry?

Reply:
The VHDL needs those numbers to calculate things and it does not have access to any ESSources or ESProducers. My strategy is to specify these numbers identically in VHDL and c++ and to check in c++ if they are consistent with tracker geometry etc. (where I must say I check not everything so far) If I would use in VHDL some hard coded numbers and in c++ the cms data bases I would not be able to catch inconsistencies as soon as those data bases do change.

Reply:
1) Where is consistency checking with the DB numbers done? It might make sense to have a single class to check this for all the geometry numbers  in ProducerSetup_cfi.py?
2) Why is the max. representable stub z uncertainty set to 30 cm? As the 2S strips are 5cm long, the z uncertainty can never exceed 5/sqrt(12) cm. And the min. stub z uncertainty of 0.1cm is almost identical to the PS pitch in z, so may not be negligeable.

Reply:
1. not sure if I check those atm, I do remember to check b field for example during Setup construction.
2. the z uncertainty I use is not detector resolution, it is detector resolution times sqrt(12) + additional terms. In the forwards region one gets a projected z uncertainty of 5cm * sinh(2.4) which is almost 30 cm

Reply:
Please check here and in code that if something is sqrt(12) * resolution it says so.

Reply:
since I don't use sqrt(12) anywhere that should be fine.

Reply:
Your reply says that mindZ & maxdZ are "resolution * sqrt(12) + terms", whereas the comment next to these cfg params says they represent the (maxy) uncertainty, (where uncertainty usually means the same as resolution). So the comment is misleading, since it misses the sqrt(12).

Reply:
uncertainty and resolution are words that every physicist knows the meaning of. "acceptance" in this context is vague. Personally I'd recommend a couple of sqrt(12). Otherwise people checking your maths in future may wonder what is going on. You don't need many of them. Just here in the cfg and in the Stub class in KFin.

Reply:
done